QuizApp/Core/NetworkServiceProtocol.swift

```diff
@@ -0,0 +1,12 @@
+import Combine
+import Reachability
+
+protocol NetworkServiceProtocol {
+
+    var networkState: CurrentValueSubject<Reachability.Connection, Never> { get }
```

Comment:
Never expose `*Subject` because it allows outsiders to emit values to it. Instead, create a computed property 
```
var networkState: AnyPublisher<Reachability.Connection, Never> {
   networkStateSubject.eraseToAnyPublisher()
}
```
and just expose `var networkState: AnyPublisher<Reachability.Connection, Never> { get }` here.